File: src/services/Auth/Auth.API/Models/NewAndUpdateAccountModel.cs
using System.ComponentModel.DataAnnotations;

namespace Auth.API.Models
{
  public record NewAndUpdateAccountModel
  {
    [Required(ErrorMessage = "Nome do usuário está vazio.")]
    public string? Nome { get; set; }

    [Required(ErrorMessage = "Nome de usuário está vazio.")]
    public string? Username { get; set; }

    [Required(ErrorMessage = "Email do Usuário está vazio.")]
    [EmailAddress(ErrorMessage = "Endereço de email inválido.")]
    public string? Email { get; set; }

    [Required(ErrorMessage = "Senha está vazio.")]
    [StringLength(50, ErrorMessage = "Senha deve ter entra 4 a 50 caracteres.", MinimumLength = 4)]
    public string? Password { get; set; }

    [StringLength(50, ErrorMessage = "Confirmação de senha deve ter entra 4 a 50 caracteres.", MinimumLength = 4)]
    [Compare("Password", ErrorMessage = "Confirmação de senha não confere.")]
    public string? ConfirmPassword { get; set; }

    [Required(ErrorMessage = "Telefone está vazio.")]
    [StringLength(17, ErrorMessage = "Telefone deve ter entra 11 a 17 caracteres.", MinimumLength = 11)]
    public string? Telefone { get; set; }

    public bool IsActive { get; set; }

    public bool IsAdmin { get; set; }
  }

  public record NewAccountResponseModel
  {
    public string Id { get; set; }

    public NewAccountResponseModel(string id)
    {
      Id = id;
    }
  }
}

File: src/mobile/lib/app/utils/models/account/new_and_update_account_model.dart
class NewAndUpdateAccountModel {
  NewAndUpdateAccountModel({
    required this.nome,
    required this.username,
    required this.email,
    required this.password,
    required this.confirmPassword,
    required this.telefone,
    this.fotoUrl,
    required this.isAtivo,
    required this.isAdmin,
  });
  late final String nome;
  late final String username;
  late final String email;
  late final String password;
  late final String confirmPassword;
  late final String telefone;
  String? fotoUrl;
  late final bool isAtivo;
  late final bool isAdmin;

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['nome'] = nome;
    data['username'] = username;
    data['email'] = email;
    data['password'] = password;
    data['confirmPassword'] = confirmPassword;
    data['telefone'] = telefone;
    data['fotoUrl'] = fotoUrl;
    data['isAtivo'] = isAtivo;
    data['isAdmin'] = isAdmin;
    return data;
  }
}

class NewAccountResponseModel {
  NewAccountResponseModel({required this.id});
  late final String id;

  NewAccountResponseModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
  }
}
